coap-core/src/main/java/com/mbed/coap/server/CoapServerBuilder.java

```diff
@@ -184,17 +188,29 @@ protected CoapUdpMessaging buildCoapMessaging() {
             server.setSpecialCoapTransactionPriority(blockTransferPriority);
             server.setTransmissionTimeout(transmissionTimeout);
 
-            server.init(duplicationMaxSize,
-                    duplicateMsgCleanIntervalMillis,
-                    duplicateMsgWarningMessageIntervalMillis,
-                    duplicateMsgDetectionTimeMillis,
-                    scheduledExecutorService,
-                    isSelfCreatedExecutor,
-                    midSupplier,
-                    maxQueueSize,
-                    defaultTransactionPriority,
-                    delayedTransactionTimeout,
-                    duplicatedCoapMessageCallback);
+            if (duplicateDetectionCache != null) {
```

Comment:
You can use the same init() method if you would make the default cache before, like

```
            if (duplicateDetectionCache == null) {
                duplicateDetectionCache = new CacheImpl<>(
                    "Default cache",
                    duplicationListSize,
                    duplicateMsgCleanIntervalMillis,
                    duplicateMsgWarningMessageIntervalMillis,
                    scheduledExecutor);
            }
```

Reply:
fixed.